File: algo/lc.35.cc
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if (nums.empty()) return 0;
        
        int cursor = 0;
        while (cursor < nums.size() && nums[cursor] <= target) {
            if (nums[cursor] == target) return cursor;
            cursor++;
        }
        return cursor;
    }
};

File: algo/lc.35.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        # x in list(int) : O(n)
        # we'd better solve this within a single pass
        if len(nums)==0: return 0
        if target <= nums[0]:
            return 0
        if len(nums)==1: return 1
        for i in range(1,len(nums)):
            a, b = nums[i-1], nums[i]
            if target <= b: return i
        return len(nums)
